internal/peer/node/start.go

```diff
@@ -828,9 +834,32 @@ func serve(args []string) error {
 		}()
 	}
 
+	signalFunc := func() {
+		// notify all externally built chaincodes of signal
+		close(externalCCSignalChan)
+
+		// wait for a message from each externally built cc that
+		// has been launched that it has terminated (and cleaned
+		// up after itself) properly
+		for ccid, instance := range containerRouter.Containers() {
```

Comment:
Going back to my other context comment, if we wired with a parent context, we could cancel everything in one shot  instead of looping.

An even cleaner implementation would be a `Shutdown`, `Kill`, or `Stop` implementation on the router that handles the details of all of this. We would want that covered with a timer to ensure we don't wait too long for cleanup.